Owncloud iOs Client/Login/Keychain/OCKeychain.m

```diff
@@ -275,17 +275,38 @@ + (BOOL) updateAllKeychainItemsToUseTheLockProperty {
     return output;
 }
 
++ (BOOL) updateAllKeychainItemsToUseAccessibleAlwaysProperty {
+    
+    BOOL output = NO;
+    
+    NSMutableDictionary *keychainQuery = [NSMutableDictionary dictionary];
+    
+    [keychainQuery setObject:(__bridge id)(kSecClassGenericPassword) forKey:(__bridge id)kSecClass];
+    [keychainQuery setObject:[UtilsUrls getFullBundleSecurityGroup] forKey:(__bridge id)kSecAttrAccessGroup];
+    
+    NSMutableDictionary *attrToUpdate = [NSMutableDictionary dictionary];
+    [attrToUpdate setObject:(__bridge id)(kSecAttrAccessibleAlways) forKey:(__bridge id)kSecAttrAccessible];
+    
+    OSStatus stsUpd = SecItemUpdate((__bridge CFDictionaryRef)(keychainQuery), (__bridge CFDictionaryRef)(attrToUpdate));
+    DLog(@"(updateLockPropertyToAccessibleAlways)Error Code: %d (0 = success)", (int)stsUpd);
+    if (stsUpd == errSecSuccess) {
+        output = YES;
+    }
+    
+    return output;
+}
+
 #pragma mark - keychain updates after some db updates
 
 #pragma mark - used to update from db version 9to10, from db to keychain
-+(BOOL)storeCredentialsOfUserFromDBVersion9To10:(UserDto *)user {
++(BOOL) storeCredentialsOfUserFromDBVersion9To10:(UserDto *)user {
     
     return [OCKeychain storeCredentials:user.credDto migratingFromDB9to10:YES];
 }
 
 #pragma mark - used to update from db version 22to23
 
-+ (BOOL)updateAllKeychainItemsFromDBVersion21or22To23ToStoreCredentialsDtoAsValueAndAuthenticationType {
++ (void) updateAllKeychainItemsFromDBVersion22To23ToStoreCredentialsDtoAsValueAndAuthenticationTypeWithCompletion:(void(^)(BOOL isUpdated))completion {
```

Comment:
Maybe if your function name has an _and_ statement could be splitted into two functions. This is better for the readability of the code and also for debugging.

Reply:
In this case what this method does it is just on action, update previous keychain items to store now a full credentials dto that also has authentication type instead of only the password

Reply:
Renamed with more accuracy description.
woo in love with the new refactor in Xcode 9 xD, @davivel you should try it, awesome.